src/Scientist/Result.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace GitHub
+{
+    /// <summary>
+    /// Provides an observed result.
+    /// </summary>
+    public interface IResult
```

Comment:
I don't think we need to introduce this interface. This is just an object that contains values (no real behavior) and it's easy to instantiate.